src/index.js

```diff
@@ -17,76 +19,74 @@ export default function({ types: t }) {
+          })
 
-            // foo.bar -> bar
-            if (t.isMemberExpression(displayName)) {
-              displayName = displayName.property
-            }
+          // foo.bar -> bar
+          if (t.isMemberExpression(displayName)) {
+            displayName = displayName.property
+          }
 
-            // identifiers are the only thing we can reliably get a name from
-            if (!t.isIdentifier(displayName)) {
-              displayName = undefined
-            } else {
-              displayName = displayName.name
-            }
+          // styled call without variable
+          if (!target) {
+            target = getTarget(path.node.tag)
           }
 
-          let newNode
-          if (addIdentifier) {
-            id++
-            // Prefix the identifier with a character if no displayName exists because CSS classes cannot start with a number
-            const identifier = `${displayName || 's'}-${hash(`${id}${displayName}`)}`
-            if (!addDisplayName) {
-              newNode = buildNodeWithIdentifier({
-                VALUE: path.node,
-                IDENTIFIER: t.stringLiteral(identifier),
-              })
-            } else {
-              newNode = buildNodeWithDisplayNameAndIdentifier({
-                VALUE: path.node,
-                DISPLAYNAME: t.stringLiteral(displayName),
-                IDENTIFIER: t.stringLiteral(identifier),
-              })
-            }
-          } else if (addDisplayName) {
-            newNode = buildNodeWithDisplayName({
-              VALUE: path.node,
-              DISPLAYNAME: t.stringLiteral(displayName),
-            })
+          // identifiers are the only thing we can reliably get a name from
+          if (!t.isIdentifier(displayName)) {
+            displayName = undefined
+          } else {
+            displayName = displayName.name
           }
-          path.node._styledComponentsSeen = true
 
-          if (!newNode) return
-          path.replaceWith(newNode)
+          id++
+          // Prefix the identifier with a character if no displayName exists because CSS classes cannot start with a number
+          const identifier = `${displayName || 's'}-${hash(`${id}${displayName}`)}`
+          // Put together the final code again
+          // Create the styled({ }) call
+          const call = buildStyledCall({
+            TARGET: target,
+            DISPLAYNAME: (addDisplayName && t.stringLiteral(displayName)) || t.identifier('undefined'),
+            IDENTIFIER: (addIdentifier && t.stringLiteral(identifier)) || t.identifier('undefined')
+          })
+          // Put together the styled call with the template literal
+          // to get the finished styled({ })`` form! 🎉
+          const styledCall = t.taggedTemplateExpression(
+            call.expression,
+            path.node.quasi
+          )
+          path._styledComponentsSeen = true
+
+          path.replaceWith(styledCall)
```

Comment:
As we got rid of wrapping, code above can be reduced to 
```
          // Put together the styled call with the template literal
          // to get the finished styled({ })`` form! 🎉
          path.node.tag = call.expression
```

No need to mark with _styledComponentsSeen